data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -282,18 +282,21 @@ private void addPart(List<String> parts, final String str, final int start, fina
     }
 
     public int findInStartGroup(final String str, int idx) {
```

Comment:
Does this need to be public?

Also, let's make `idx` final to avoid errors in changing this.